docs/adrs/0009-nimbus-web-sdk-architecture.md

```diff
@@ -0,0 +1,97 @@
+  * The SDK needs to support Python
+  * The SDK needs to be stateless
+
+## Considered Options
+
+* Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+* Cargo features, one library
+* Cargo features, separate library
+* Expose a stateless API alongside the stateful API
+
+## Decision Outcome
+
+[justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+## Pros and Cons of the Options
+
+### Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+
+For this option, we would take the core Nimbus business logic currently in the `nimbus-sdk` library and move it to a new `nimbus-core` library. 
+The `nimbus-sdk` and `cirrus-sdk` libraries would depend on the `nimbus-core` library, extending the functionality for their various use cases. 
+
+* Good, because separate libraries allow for strong separation of concerns.
+* Bad, because defining and using types becomes more difficult.
+  * Traits will be needed to extend functionality in the Nimbus Core library. While dynamic typing with traits is supported, it is one of the more complex features of Rust.
+* Bad, because it would result in a significant amount of code churn.
+
+### Cargo features, one library
+
+Cargo features allow for conditional compilation of libraries and library dependencies. 
+Native Rust attributes are applied to sections of Rust code to conditionally compile structs, methods, variables, and more.
+
+For more information, check out the Cargo Features reference[<sup>[1.ii]</sup>](#links) available in the Rust Cargo book.
+
+* Good, because the use of feature configurations is familiar to Rust developers.
+* Good, because all code relating to Nimbus business logic remains in one place.
+* Good, because UniFFI bindings can be conditionally generated for each feature with different UDLs.
+* Good, because dependencies can be excluded based on enabled features.
+  * As an example, RKV will only be needed for Nimbus, so the RKV dependency will only be included in compilation when the `nimbus` feature is enabled.
+* Good, because it will not increase the size of the existing Nimbus SDK binary.
+* Bad, because onboarding developers learning Rust would likely have a more difficult time understanding Cargo features.
+
+### Cargo features, separate library
+
+This is roughly the same concept as above, but the features will be used in a different way.
+Instead of features for both Nimbus and Cirrus, we would include a `stateless` feature in the Nimbus library to disable state-oriented features and dependencies.
+The Cirrus library will then exist as a separate library from the Nimbus library.
```

Comment:
It's not clear to me what the role of a separate Cirrus library would be in this scenario?

Reply:
In this scenario the purpose of the Cirrus library would be a separation of concerns re: the Cirrus UniFFI UDL and exported Rust/Python Cirrus-specific functions. These would likely primarily be helper methods, e.g: parsing JSON from Python before getting to the core function or other things along those lines. 

The Cirrus SDK would import the Nimbus library as a dependency, enabling the `stateless` feature to cut out dependencies like RKV/Nimbus-specific UniFFI and state-oriented features such as the event store and internal database.

Reply:
Makes sense, can you add some of that explanation back to the ADR? 🙏

Reply:
Sure!